crates/matrix-sdk-ui/tests/integration/timeline/pinned_event.rs

```diff
@@ -355,20 +599,56 @@ impl TestHelper {
 
     async fn setup_sync_response(
         &mut self,
-        text_messages: Vec<(&str, &str, bool)>,
+        text_messages: Vec<(MessageType, bool)>,
         pinned_event_ids: Option<Vec<&str>>,
     ) -> Result<SyncResponse, matrix_sdk::Error> {
         let mut joined_room_builder = JoinedRoomBuilder::new(&self.room_id);
-        for (id, txt, add_to_timeline) in text_messages {
-            let event_id: OwnedEventId = id.try_into().unwrap();
-            let f = EventFactory::new().room(&self.room_id);
-            let event_builder = f.text_msg(txt).event_id(&event_id).sender(*BOB);
-            mock_event(&self.server, &self.room_id, &event_id, event_builder.into_timeline()).await;
+        for (message_type, add_to_timeline) in text_messages {
+            let (event_id, timeline_item) = match message_type {
```

Comment:
nit: not `timeline_item`, but `timeline_event`, I think; item should be reserved to the UI Timeline object :)